BeSpoke/Views/ScrollingView.swift

```diff
@@ -0,0 +1,104 @@
+//
+// ScrollingView.swift
+// bespoke
+//
+// Created by Ethan Wright on 2023-12-11
+//
+        
+
+import Foundation
+import FirebaseFirestore
+import SwiftUI
+
+struct ScrollingView: View {
+    
+    // This is fake data. If you couldn't tell.
+    let quote = Quote(
+        title: "Quote of the day",
+        quote: "Insert Knowledge here, perhaps why there's still white backgrounds.")
+    
+    let tags: [Tag] = [
+        Tag(
+            notes: [1, 2],
+            title: "smiley",
+            imagePath: "face.smiling"),
+        Tag(
+            notes: [1, 3, 5],
+            title: "sad",
+            imagePath: "hand.thumbsdown"),
+        Tag(
+            notes: [2, 3, 4],
+            title: "rainy",
+            imagePath: "cloud"),
+        Tag(
+            notes: [4],
+            title: "sunny",
+            imagePath: "sun.min"),
+        Tag(
+            notes: [5],
+            title: "mid",
+            imagePath: "stop.fill"),
+        Tag(            notes: [],
+            title: "best day",
+            imagePath: "music.note")
+    ]
+    
+    // Real code follows.
+
+    @State private var activeUser: User = getActiveUser()
+    @State private var notes: [Note] = []
+
+    var body: some View {
+        NavigationView {
+            ScrollView {
+                QuoteView(quote: quote)
+                LazyVGrid(columns: [GridItem(.flexible())]) {
+                    ForEach(notes) { note in
+                        // TODO: tag matching
+                        NoteView(note: note, tags: tags)
+                    }
+                }
+                .onAppear(perform: fetchDocs)
+            }
+            .navigationTitle("Bespoke")
+            .toolbar {
+                ToolbarItem(placement: .navigationBarTrailing) {
+                    NavigationLink(destination: ProfileView(user: $activeUser)) {
+                        Image(systemName: "person.circle")
+                            .resizable()
+                            .frame(width: 30, height: 30)
+                            .foregroundColor(.blue)
+                    }
+                }
+            }
+            .background(Color(red: 0.0, green: 0.1, blue: 0.0))
+            .preferredColorScheme(.dark)
+        }
+    }
+    
+    func fetchDocs() {
+        let db = Firestore.firestore()
```

Comment:
Does this ever fail  - e.g. network connection problem? If so, could make a separate 'firestone utilities' file which manages error handling from stuff like that (I'm assuming `let db = Firestore.firestore()` is meant to be called every access rather than having a global initialized once?)

Reply:
Would prefer inheritance over utility classes if possible, but I've never seen this wrapped in a `try` and Swift is pretty hostile with forcing those. I'm gonna leave this for now, but it may be possible to call this once rather than every time you want to open the database.